posthog/models/insight.py

```diff
@@ -108,11 +108,11 @@ def dashboard_filters(self, dashboard: Optional[Dashboard] = None):
 
     @property
     def effective_restriction_level(self) -> Dashboard.RestrictionLevel:
-        dashboards = list(self.dashboards.all())
-        if not dashboards:
+        tiles = list(self.dashboard_tiles.all())
+        if not tiles:
             return Dashboard.RestrictionLevel.EVERYONE_IN_PROJECT_CAN_EDIT
 
-        restrictions = [d.effective_restriction_level for d in dashboards]
+        restrictions = [tile.dashboard.effective_restriction_level for tile in tiles]
```

Comment:
Why the switch from `self.dashboards` to `self.dashboard_tiles`?
Extra question: Would `.select_related('dashboard')` optimize the `tiles` query?